File: eureka/android/library/src/main/java/com/lifeplus/Pojo/Enum/BleProcEnum.java
package com.lifeplus.Pojo.Enum;

public enum BleProcEnum {
    CONNECT("Connect", 60),
    INSTANT_MEASURE("Measure", 90),
    AUTO_MEASURE("Measure", 90),
	STEP_COUNT("Stepcount", 60),
    CALIBRATE("Calibration", 180),
    APP_SYNC("App Sync", 60),
    NONE("None", 0),
    MEAL_DATA("MealData", 60),
    READ_RAW_DATA("RawData", 60),
    NOTIFICATION("notification", 60),
    FIRMWARE_UPDATE("FirmwareUpdate", 0),
    STEP_GOAL_UPDATE("Step goal update", 0),
    TIMEZONE_UPDATE("Timezone update", 0);

    private final String _desc;
    private final int _timeoutPeriod;       // in seconds

    BleProcEnum(String pDesc, int pTimeoutPeriod) {
        _desc = pDesc;
        _timeoutPeriod = pTimeoutPeriod;
    }

    public String getDesc() {
        return _desc;
    }

    public int getTimeoutMillis() {
        return _timeoutPeriod * 1000;
    }
}

File: eureka/ios/nativeLib/nativeLib/Models/Enum/BleProcEnum.swift
//
//  BleProcEnum.swift
//  BLEDemo
//

import Foundation

public enum BleProcEnum: String, CaseIterable {
    
  case CONNECT = "Connect"
  case INSTANT_MEASURE = "Measure"
  case AUTO_MEASURE = "Auto Measure"
  case STEP_COUNT = "Stepcount"
  case CALIBRATE = "Calibration"
  case APP_SYNC = "App Sync"
  case NONE = "None"
  case MEAL_DATA = "MealData"
  case READ_RAW_DATA = "RawData"
  case AUTO_MEASURE_SYNC = "Auto Measure Sync"
  case FIRMWARE_UPDATE = "Firmware Update"
  case STEP_GOAL_UPDATE = "Step goal update"
  case TIMEZONE_UPDATE = "Timezone update"
   
  public var desc:String {
    switch self {
    case .CONNECT: return "Connect"
    case .INSTANT_MEASURE: return "Measure"
    case .AUTO_MEASURE: return "Measure"
    case .STEP_COUNT: return "Stepcount"
    case .CALIBRATE: return "Calibration"
    case .APP_SYNC: return "App Sync"
    case .NONE: return "None"
    case .MEAL_DATA: return "MealData"
    case .READ_RAW_DATA: return "RawData"
    case .AUTO_MEASURE_SYNC: return "AUTO MEASURE SYNC"
    case .FIRMWARE_UPDATE: return "Firmware Update"
    case .STEP_GOAL_UPDATE: return "Step goal update"
    case .TIMEZONE_UPDATE: return "Timezone update"
    }
  }
  
}
